addons/website/models/website_visitor.py

```diff
@@ -332,15 +333,25 @@ def _merge_visitor(self, target):
         self.website_track_ids.visitor_id = target.id
         self.unlink()
 
-    def _cron_unlink_old_visitors(self):
+    def _cron_unlink_old_visitors(self, auto_commit=True):
         """ Unlink inactive visitors (see '_inactive_visitors_domain' for
         details).
 
         Visitors were previously archived but we came to the conclusion that
         archived visitors have very little value and bloat the database for no
-        reason. """
+        reason.
 
-        self.env['website.visitor'].sudo().search(self._inactive_visitors_domain()).unlink()
+        :param bool auto_commit: Whether the database cursor should be committed
+                                 after each batch of unlinked inactive visitors
+        """
+        visitor_model = self.env['website.visitor']
+        for inactive_visitors_batch in split_every(
+                1000,
+                visitor_model.sudo().search(self._inactive_visitors_domain()).ids,
+                visitor_model.browse):
```

Comment:
Nitpicking: Indent seems weird but feel free to ignore.
```suggestion
        for inactive_visitors_batch in split_every(
            1000,
            visitor_model.sudo().search(self._inactive_visitors_domain()).ids,
            visitor_model.browse,
        ):
```